src/briefcase/commands/create.py

```diff
@@ -487,7 +488,48 @@ def _install_app_requirements(
         else:
             self.logger.info("No application requirements.")
 
-    def install_app_requirements(self, app: BaseConfig, test_mode: bool):
+    def _calc_app_requirements(self, app: BaseConfig, test_mode: bool, relock: bool):
+        requires = app.requires.copy() if app.requires else []
+        if test_mode and app.test_requires:
+            requires.extend(app.test_requires)
+
+        lock_attr = "requires_lock" if not test_mode else "test_requires_lock"
+        try:
+            lock_value = getattr(app, lock_attr)
+        except AttributeError:
+            return requires
+        else:
+            lock_file = self.base_path / lock_value
+
+        if relock:
+            with tempfile.NamedTemporaryFile(mode="w+") as fpout:
+                fpout.writelines(line + "\n" for line in requires)
+                fpout.flush()
+                res = self.tools[app].app_context.run(
+                    [
+                        sys.executable,
+                        "-u",
+                        "-m",
+                        "piptools",
+                        "compile",
+                        fpout.name,
+                        "--output-file=-",
+                    ],
+                    text=True,
+                    capture_output=True,
+                    check=True,
+                )
+            lock_file.write_text(res.stdout)
+
+        requires = [
+            line.split("#", 1)[0].strip() for line in lock_file.read_text().splitlines()
+        ]
+        requires = [potential for potential in requires if potential != ""]
+        return requires
+
+    def install_app_requirements(
+        self, app: BaseConfig, test_mode: bool, relock: bool = False
```

Comment:
`relock` shouldn't have a default value here; we're in a position to be explicit whenever it's used.